src/app_service/service/community/service.nim

```diff
@@ -276,6 +276,8 @@ QtObject:
 
       if response.result != nil and response.result.kind != JNull:
         let community = response.result["communities"][0].toCommunityDto()
+        # add this to the joinedCommunities list
+        self.joinedCommunities[community.id] = community
```

Comment:
I can't comment above, but on line 249, we'd need to remove from `joinedCommunities` like you did here